legend-engine-external-format-cpb/src/main/java/org/finos/legend/engine/external/format/cpb/extension/CpbGenerationExtension.java

```diff
@@ -0,0 +1,97 @@
+package org.finos.legend.engine.external.format.cpb.extension;
+
+import org.eclipse.collections.api.RichIterable;
+import org.finos.legend.engine.external.format.cpb.schema.generations.CpbGenerationConfig;
+import org.finos.legend.engine.external.format.cpb.schema.generations.CpbGenerationService;
+import org.finos.legend.engine.external.shared.format.extension.GenerationExtension;
+import org.finos.legend.engine.external.shared.format.extension.GenerationMode;
+import org.finos.legend.engine.external.shared.format.generations.description.FileGenerationDescription;
+import org.finos.legend.engine.external.shared.format.generations.description.GenerationConfigurationDescription;
+import org.finos.legend.engine.external.shared.format.generations.description.GenerationProperty;
+import org.finos.legend.engine.external.shared.format.imports.description.ImportConfigurationDescription;
+import org.finos.legend.engine.language.pure.compiler.toPureGraph.CompileContext;
+import org.finos.legend.engine.language.pure.compiler.toPureGraph.PureModel;
+import org.finos.legend.engine.language.pure.modelManager.ModelManager;
+import org.finos.legend.engine.protocol.pure.v1.model.packageableElement.PackageableElement;
+import org.finos.legend.engine.protocol.pure.v1.model.packageableElement.fileGeneration.FileGenerationSpecification;
+import org.finos.legend.pure.generated.*;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class CpbGenerationExtension implements GenerationExtension
```

Comment:
A question: how feasible to generalize this to PersistenceContextGenerationExtension? Would love to see if shell can be reused for all persistence contexts with internal support for switching between based on a property.